src/librustc_mir/borrow_check/mod.rs

```diff
@@ -140,6 +141,20 @@ fn do_mir_borrowck<'a, 'gcx, 'tcx>(
         .as_local_hir_id(def_id)
         .expect("do_mir_borrowck: non-local DefId");
 
+    let mut local_names = IndexVec::from_elem(None, &input_mir.local_decls);
+    for var_debug_info in &input_mir.var_debug_info {
+        if let Place::Base(PlaceBase::Local(local)) = var_debug_info.place {
```

Comment:
Can we even perform that optimization without dedicated analysis? (It's what you get from my NRVO optimization, I think, perhaps with the SROA one too).

And we would care, because MIR borrowck needs to report errors about user variables (which `a` and `b` are).

The only non-local debuginfo during MIR borrowck, right now, are upvars.